llvm/unittests/TargetParser/TargetParserTest.cpp

```diff
@@ -1012,11 +1013,18 @@ TEST(TargetParserTest, getARMCPUForArch) {
 
 TEST(TargetParserTest, ARMPrintSupportedExtensions) {
   std::string expected = "All available -march extensions for ARM\n\n"
-                         "\tcrc\n\tcrypto\n\tsha2";
+                         "    Name                Description\n"
+                         "    crc                 This is a long dummy description\n"
+                         "    crypto              This is a long dummy description\n"
+                         "    sha2                This is a long dummy description\n";
+
+  llvm::StringMap<llvm::StringRef> DummyMap;
+  for (const auto &E : llvm::ARM::ARCHExtNames)
+    DummyMap.insert({E.Name, "This is a long dummy description"});
```

Comment:
This is ok but what I meant specifically was, add a dummy description for just one extension. Then these tests are covering the case where we do and don't have an extension.

Reply:
The way you tested RISCV previously in unittest (in full-test) require to fill up all fields.
I don't see harm to go for all fields.

Reply:
The harm is we aren't testing what happens if an extension has no description. If the underlying implementation changes in the future to something that does fault on a missing description, no one will know until a user tries the option.

Reply:
Also it saves you adding a method to RISC-V just to get the list of extensions. Which we did for AArch64 but it was a hack there too that I would love to also remove one day.

Reply:
As method, isn't straight possible in the way this RISCV class is now.

Reply:
Not sure what you mean.
```
DummyMap["crc"] = "This is a long dummy description";
```
Is what I am suggesting. Adjust the expected output as needed.

Do that for RISC-V and AArch64 too and there's no need for an extra method in RISC-V to get all the extension names.